src/test/ui/error-codes/e0119/issue-28981.rs

```diff
@@ -1,3 +1,10 @@
+// only-x86_64
+
+// FIXME: Depending on the order of `non_blanket_impls`, this test
+// mentions a different impl for the overlap.
+//
+// The order of these impls isn't currently stable between different targets,
+// so only test this on one of them.
```

Comment:
Based on where it fails, I suspect it may have something to do with how [`memchr` uses a build script to have per-platform behavior](https://github.com/BurntSushi/memchr/blob/8e1da98fee06d66c13e66c330e3a3dd6ccf0e3a0/build.rs).

I wonder if that results in different `-Cmetadata` and therefore different `StableCrateId`s.

Ideally we would not show random `impl`s in indirect dependencies, if direct dependencies themselves have conflicts (i.e. prefer `std` - though ideally it would be `core`).

Alternatively, we can try to rely on stability, and prefer conflicts with stable `impl`s.
Worst case, we buffer up all the errors, and sort them alphabetically or something, listing the first 4 or so (like other errors do). It's an error case, so the extra cost should be worth it.

cc @nikomatsakis (I'm surprised coherence errors can be so order-sensitive)

Reply:
we only get these fairly unhelpful errors if our impl is illegal and fails the coherence check.
imo only emitting the following error is the best fix here:
```
error[E0210]: type parameter `Foo` must be used as the type parameter for some local type (e.g., `MyStruct<Foo>`)
  --> $DIR/issue-28981.rs:5:6
   |
LL | impl<Foo> Deref for Foo { }
   |      ^^^ type parameter `Foo` must be used as the type parameter for some local type
   |
   = note: implementing a foreign trait is only possible if at least one of the types for which it is implemented is local
   = note: only traits defined in the current crate can be implemented for a type parameter
```
so this overlapping impl errors would only trigger for either an actually relevant extern impl, e.g. `impl<T> Trait for Vec<T>` and `impl Trait for Vec<LocalType>`, or when both impls are in the local crate, at which point "nondeterminism" isn't an issue.

will look into this after my vacation next week.